### PR TITLE
Fix convolve for non-float masked kernels

### PR DESCRIPTION
**Pull request by jamienoss**
Signed-off-by: James Noss <jnoss@stsci.edu>

### CODE REVIEW

File: astropy/convolution/convolve.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         array_internal[mask != 0] = np.nan
     if np.ma.is_masked(kernel):
         # *kernel* doesn't support NaN interpolation, so instead we just fill it
-        kernel_internal = kernel.filled(fill_value)
+        kernel_internal = kernel_internal.filled(fill_value)
```
Comment (jamienoss) [inline astropy/convolution/convolve.py]: If ``kernel`` is __not__ a float, it's previous conversion to a ``float`` array ([here](https://github.com/astropy/astropy/pull/7542/files#diff-f632180d9f1e3fcf689f62e4c424e94fR175)) gets undone as ``kernel.filled()`` returns a copy of the original ``kernel`` type.

**Comment by jamienoss** [comment]
https://travis-ci.org/astropy/astropy/jobs/388842732
```
CondaHTTPError: HTTP 504 GATEWAY TIME-OUT for url <https://repo.anaconda.com/pkgs/free/linux-64/repodata.json.bz2>
Elapsed: 00:10.007970
CF-RAY: 426c30e23d8c570b-IAD
A remote server error occurred when trying to retrieve this URL.
A 500-type error (e.g. 500, 501, 502, 503, etc.) indicates the server failed to
fulfill a valid request.  The problem may be spurious, and will resolve itself if you
try your request again.  If the problem persists, consider notifying the maintainer
of the remote server.
```

Can someone restart that test, please, cheers?

**Comment by keflavich** [comment]
done

**Comment by pllim** [comment]
Please add a change log for 2.0.8 under bug fixes.

@bsipocz , if backport to 2.x is not possible for this, please remilestone.

**Comment by pllim** [comment]
Since it is approved by maintainer and tests passed, I am merging this. Thanks!

**Pull request merged**